src/parser/name_resolution.c

```diff
@@ -1201,74 +1213,78 @@ pt_bind_scope (PARSER_CONTEXT * parser, PT_BIND_NAMES_ARG * bind_arg)
 	  else if (table->node_type == PT_DBLINK_TABLE)
 	    {
 	      assert (spec->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE);
+
+	      int err;
+	      S_REMOTE_TBL_COLS *rmt_tbl_cols;
+	      REMOTE_COLS *remote;
+	      PT_DBLINK_INFO *dblink_table = &table->info.dblink_table;
+
 	      if (table->info.dblink_table.is_name && table->info.dblink_table.url == NULL)
 		{
-		  int err;
-		  S_REMOTE_TBL_COLS *rmt_tbl_cols;
-		  REMOTE_COLS *remote;
-		  PT_DBLINK_INFO *dblink_table = &table->info.dblink_table;
-
 		  if (pt_resolve_dblink_server_name (parser, table, NULL) != NO_ERROR)
 		    {
 		      return;
 		    }
+		}
 
-		  /* remote table's column list */
-		  rmt_tbl_cols = new (std::nothrow) S_REMOTE_TBL_COLS;
+	      /* remote table's column list */
+	      rmt_tbl_cols = new (std::nothrow) S_REMOTE_TBL_COLS;
```

Comment:
You should check for memory allocation failures.
Please add at least assert(rmt_tbl_cols != NULL)

Reply:
I fixed it.